### PR TITLE
[siminstaller] Adjust according to recent Xcode changes.

### PR DESCRIPTION
**Pull request by rolfbjarne**
The url to download the list of simulators that are available has changed, and
the format changed slightly as well.

### CODE REVIEW

**Review by mandel-macaque — APPROVED** [review]

**Review by chamons — APPROVED** [review]

File: tools/siminstaller/Program.cs [hunk 1]
```diff
@@ -200,7 +229,14 @@ public static int Main (string [] args)
 
 				var name = Replace (nameNode.InnerText, dict);
 				var source = Replace (sourceNode.InnerText, dict);
-				var installPrefix = Replace (installPrefixNode.InnerText, dict);
+				var installPrefix = Replace (installPrefixNode?.InnerText, dict);
+
+				if (installPrefix is null) {
+					// This is just guesswork
+					var simRuntimeName = name.Replace (" Simulator", ".simruntime");
+					installPrefix = $"/Library/Developer/CoreSimulator/Profiles/Runtimes/{simRuntimeName}";
```
Comment (rmarinho) [inline tools/siminstaller/Program.cs:237]: What should be the path created? i ported this to xharness but it's not working .. i m getting a path with a space not sure if it's right.. 

```
 Oct  3 10:52:12  installer[5885] <Info>: PackageKit: Registered bundle file:///Library/Developer/CoreSimulator/Profiles/Runtimes/iOS%2014.5.simruntime/Contents/Resources/RuntimeRoot/Applications/AMSEngagementViewService.app/ for uid 0
````

Comment (rolfbjarne) [reply tools/siminstaller/Program.cs:237]: This logic might be wrong for the iOS 14.5 simulator, we only install the iOS 12.4 simulator (which works for us).

The space (`%20`) looks normal to me:

```
$ ls -la /Library/Developer/CoreSimulator/Profiles/Runtimes/
total 0
drwxr-xr-x  32 root  admin  1024 Sep 30 17:27 .
drwxr-xr-x   3 root  admin    96 Feb  8  2019 ..
drwxrwxr-t   3 root  admin    96 Jun  7 08:50 iOS 10.0.simruntime
drwxrwxr-t   3 root  admin    96 Jun  7 08:50 iOS 10.1.simruntime
drwxrwxr-t   3 root  admin    96 Jun  7 08:52 iOS 10.2.simruntime
drwxrwxr-t   3 root  admin    96 Jun  7 08:51 iOS 10.3.simruntime
drwxrwxr-t   3 root  admin    96 Jun  7 08:51 iOS 11.0.simruntime
drwxrwxr-t   3 root  admin    96 Jun  7 08:51 iOS 11.4.simruntime
drwxrwxr-t   3 root  admin    96 Jun  7 08:50 iOS 12.0.simruntime
drwxrwxr-t   3 root  admin    96 Jun  7 08:51 iOS 12.1.simruntime
[...]
```

@akoeplinger pointed out that Xcode does something different with newer simulators (mounts a dmg in `/Library/Developer/CoreSimulator/Volumes/`) - but I have no idea how to download the dmg and make Xcode mount it.

Comment (rmarinho) [reply tools/siminstaller/Program.cs:237]: Yeah but it didn't work for us with iOS 12.4 .. Did you try on a new machine works for you ?

```
Process ID: 6209
[1.0.0-prerelease.22480.2+db24a6d8a7fd009688425151e3da19cb5c4ca47f] XHarness command issued: apple test --app=/Users/runner/work/1/s/src/Controls/tests/DeviceTests/bin/Release/net6.0-ios/iossimulator-x64/Microsoft.Maui.Controls.DeviceTests.app --targets=ios-simulator-64_12.4 --output-directory=/Users/runner/work/1/a/test-results --verbosity=Debug
info: Preparing run for ios-simulator-64_12.4
info: Looking for available ios-simulator-64_12.4 simulators..
dbug: Looking for available ios-simulator-64_12.4 simulators. Storing logs into list-ios-simulator-64_12.4-20221003_105020.log
fail: Failed to find/create suitable simulator
XHarness exit code: 81 (DEVICE_NOT_FOUND)
An error occurred when executing task 'Test'.
```

I installed via Xcode 13.7 and u see it there

```
ruimarinho@iMSFT-MacBookPro Runtimes % ls -la /Library/Developer/CoreSimulator/Profiles/Runtimes/
total 0
drwxr-xr-x  3 root  wheel  96 Oct  3 12:28 .
drwxr-xr-x  3 root  wheel  96 Oct  3 12:28 ..
drwxrwxr-t  3 root  admin  96 Oct  3 12:23 iOS 13.7.simruntime
```

**Pull request merged**